pkg/sql/logictest/testdata/logic_test/enums

```diff
@@ -410,3 +410,42 @@ CREATE TABLE t1 (
    INDEX i (x ASC),
    FAMILY "primary" (x, rowid)
 )
+
+# Test that the implicit array type has been created, and that we can use it.
+query T
+SELECT ARRAY['hello']::_greeting
+----
+{hello}
+
+statement ok
+CREATE TABLE enum_array (x _greeting, y greeting[]);
+INSERT INTO enum_array VALUES (ARRAY['hello'], ARRAY['hello']), (ARRAY['howdy'], ARRAY['howdy'])
+
+query TT rowsort
+SELECT * FROM enum_array
```

Comment:
Can you check that the elements of this column are properly enum-typed?

Reply:
what do you mean by this?

Reply:
Like - `select pg_typeof(x[0]) from enum_array` should return `greeting`, and `x[0]` should be a `greeting` and not a `string`. Not sure how you can test that. I'm sure the type system mostly guarantees this already. I think it'd be good to have a smoke test though that you can roundtrip enums through arrays, through disk, and back.

Reply:
Done, dang you caught a sneaky bug with names not getting set up in the right spots.